functions/go/apply-setters/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+//nolint
+func main() {
+	resourceList := &framework.ResourceList{}
+	resourceList.FunctionConfig = map[string]interface{}{}
+
+	cmd := framework.Command(resourceList, func() error {
+		s, err := getSetters(resourceList.FunctionConfig)
+		if err != nil {
+			return fmt.Errorf("failed to parse function config: %w", err)
+		}
+		_, err = s.Filter(resourceList.Items)
+		if err != nil {
+			return fmt.Errorf("failed to apply setters: %w", err)
+		}
+		return nil
+	})
+
+	cmd.Long = usage()
+	if err := cmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func usage() string {
+	return `Apply setter values to resource fields with setter references.
```

Comment:
Help messages will be the only comprehensive documentation (function reference). Site page will be generated from this. Think of this as the user guide document for setters. Are you planning to expand this in the future to be more comprehensive? 


@mengqiy

Reply:
We are going to create reference doc from a function's help text.
The 1st paragraph should be a concise summary in 1 or 2 sentences. This paragraph will be used as the short description in the catalog's landing page which may look like https://googlecontainertools.github.io/kpt/guides/consumer/function/catalog/transformers/.
The whole help text should be comprehensive.
I know this help text will be updated again when you implement CRD as config.

Reply:
The usage() here reads more like an example package than a comprehensive reference to this function.

Missing:
- What's a setter? Why should you use it?
- Subpackages
- Interpolation
- Failure cases
- ...

Reply:
I will take another pass on the documentation after implementing subpackages and will work with @mengqiy as to how detailed the documentation should be and maintain consistency with other function docs.